File: API/Database/DbModels/Competition.cs
using System.ComponentModel.DataAnnotations;

namespace API.Database.DbModels;

public class Competition
{
    [Key]
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Date { get; set; }
    public List<Arbiter> Arbiters { get; set; }
    public List<Result> Results { get; set; }
    public List<Rider> Riders { get; set; }
    public List<Horse> Horses { get; set; }
}

File: client/src/app/modules/administration/models/Competition.ts
import { Arbiter } from "./Arbiter";
import { RidersCompetetion } from "./RidersCompetition";

export class Competition {

    public static clone(c: Competition): Competition {
        return new Competition(c.id, c.name, c.description, c.date,
            c.riders.map(rider => RidersCompetetion.cloneFromRider(rider)),
            c.arbiters.map(arbiters => Arbiter.clone(arbiters)))
    }

    constructor(
        public id: number,
        public name: string,
        public description: string,
        public date: any,
        public riders: RidersCompetetion[] = [],
        public arbiters: Arbiter[] = [],
        public ridersIds?: number[],
        public arbitersIds?: number[]
    ) { }
}